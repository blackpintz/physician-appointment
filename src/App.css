.form {
  height: 100%;
}

.physician {
  height: 50vh;
}
