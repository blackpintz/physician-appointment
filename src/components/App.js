import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';
import '../App.css';
import Login from './Login';
import AppointmentList from './AppointmentList';
import Home from './Home';
import Physician from './Physician';
import currentUser from '../helpers/currentUser';
import Signup from './Signup';
import Menu from './Menu';

const App = () => (
  <>
    <Menu />
    <Switch>
      <Route exact path="/login" render={routeProps => (currentUser() ? <Redirect to="/" /> : <Login routeProps={routeProps} />)} />
      <Route exact path="/">
        {currentUser() ? <Home /> : <Redirect to="/login" />}
      </Route>
      <Route exact path="/signup" render={routeProps => (currentUser() ? <Redirect to="/" /> : <Signup routeProps={routeProps} />)} />
      <Route exact path="/appointments">
        {currentUser() ? <AppointmentList /> : <Redirect to="/login" />}
      </Route>
      <Route exact path="/physician/:id" render={routeProps => (!currentUser() ? <Redirect to="/" /> : <Physician id={routeProps.match.params.id} routeProps={routeProps} />)} />
    </Switch>
  </>
);

export default App;
