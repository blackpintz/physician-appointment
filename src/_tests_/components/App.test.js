import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json';
import App from '../../components/App';

Enzyme.configure({ adapter: new Adapter() });

test('renders App correctly', () => {
  const wrapper = shallow(<App />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
