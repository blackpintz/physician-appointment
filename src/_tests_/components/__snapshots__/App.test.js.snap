// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<Fragment>
  <withRouter(Menu) />
  <Switch>
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/"
    >
      <Redirect
        to="/login"
      />
    </Route>
    <Route
      exact={true}
      path="/signup"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/appointments"
    >
      <Redirect
        to="/login"
      />
    </Route>
    <Route
      exact={true}
      path="/physician/:id"
      render={[Function]}
    />
  </Switch>
</Fragment>
`;
